video/audio_channel.h

```diff
@@ -264,46 +271,88 @@ void audio_channel::setFrequency(uint16_t frequency) {
 	}
 }
 
-void audio_channel::setVolumeEnvelope(std::unique_ptr<VolumeEnvelope> envelope) {
+void AudioChannel::setDuration(int32_t duration) {
+	debug_log("AudioChannel: setDuration %d\n\r", duration);
+	if (duration == 0xFFFFFF) {
+		duration = -1;
+	}
```

Comment:
duration is a 24-bit value.  if that's all 1s then that's an "indefinite" duration